cpp/src/io/parquet/reader_impl.cpp

```diff
@@ -235,44 +234,86 @@ reader::impl::impl(std::vector<std::unique_ptr<datasource>>&& sources,
                               _timestamp_type.id());
 }
 
+reader::impl::impl(std::size_t chunk_read_limit,
+                   std::vector<std::unique_ptr<datasource>>&& sources,
+                   parquet_reader_options const& options,
+                   rmm::cuda_stream_view stream,
+                   rmm::mr::device_memory_resource* mr)
+  : impl(std::forward<std::vector<std::unique_ptr<cudf::io::datasource>>>(sources),
+         options,
+         stream,
+         mr)
```

Comment:
Shouldn't this be the other way around? This constructor should be caller from the other one, with `chunk_read_limit = 0`.
That way there would be no reason to initialize `_chunk_read_limit` in the constructor body.

Reply:
Hah yes, it should be called from the other one.